### PR TITLE
feat: add command to generate all guides

### PR DESCRIPTION
**Pull request by JacquesDurand**

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -82,6 +82,11 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->arrayNode('guides')
                     ->info('Guides configuration')
                     ->children()
+                        ->scalarNode('src')
+                            ->info('Root directory for guides')
+                            ->defaultValue('guide')
```
Comment (vincentchalamon) [inline src/DependencyInjection/Configuration.php]:
```suggestion
                            ->defaultValue('guides')
```

**Review by vincentchalamon — CHANGES_REQUESTED** [review]
Please also add GuidesCommandTest

File: src/Command/GuidesCommand.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        private readonly string $defaultTemplate
+    ) {
+        parent::__construct(name: 'guides');
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Creates Guides based on PHP code')
+            ->addOption(
```
Comment (vincentchalamon) [inline src/Command/GuidesCommand.php]: Should be a required argument (cf. ReferencesCommand)

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: src/Command/GuidesCommand.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $style = new SymfonyStyle($input, $output);
+
+        $guideDir = $this->configuration->get('guides.src');
+        $template = $input->getOption('template');
+        $out = $input->getOption('output');
+
+        $referenceExtension = pathinfo(preg_replace('/\.twig$/i', '', $template), \PATHINFO_EXTENSION);
```
Comment (soyuka) [inline src/Command/GuidesCommand.php]: use https://symfony.com/doc/current/components/filesystem.html#path-manipulation-utilities instead?

File: src/Command/GuidesCommand.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        foreach ((new Finder())->files()->in($guideDir) as $file) {
+            $input = $out ?
+                new ArrayInput([
+                    'filename' => $file->getPathName(),
+                    '--output' => sprintf('%s%s%s.%s', rtrim($out, \DIRECTORY_SEPARATOR), \DIRECTORY_SEPARATOR, $file->getBaseName('.php'), $referenceExtension),
```
Comment (soyuka) [inline src/Command/GuidesCommand.php]: use https://symfony.com/doc/current/components/filesystem.html#path-manipulation-utilities instead?

File: src/Command/GuidesCommand.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                description: 'The path to the template file to use to generate each guide',
+                default: $this->defaultTemplate
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $style = new SymfonyStyle($input, $output);
+
+        $guideDir = $this->configuration->get('guides.src');
```
Comment (soyuka) [inline src/Command/GuidesCommand.php]: can there be an option for that? We take option first, then configuration right?

Comment (JacquesDurand) [reply src/Command/GuidesCommand.php]: sure, on it !

Comment (vincentchalamon) [reply src/Command/GuidesCommand.php]: Currently, ReferencesCommand does not accept an input directory, it's based on the configuration (AFAIK)

Comment (JacquesDurand) [reply src/Command/GuidesCommand.php]: This can also be done if needed ?

**Review by soyuka — APPROVED** [review]

File: tests/Fixtures/app/pdg.config.yaml [hunk 6]
```diff
@@ -11,5 +11,6 @@ api_platform_pdg:
     output: 'pages/references'
     base_url: '/pages/references'
   guides:
+    src: "guides"
```
Comment (vincentchalamon) [inline tests/Fixtures/app/pdg.config.yaml]:
```suggestion
    src: 'guides'
```

**Review by vincentchalamon — APPROVED** [review]

**Pull request merged**